### DISCUSSION TITLE
Is Nanoid RFC3986 compatible?

### DISCUSSION THREAD

**Discussion by arsekil** [ai/nanoid#503] [Q&A] [answered]
Greetings,

I was wondering if Nanoid is RFC3986 compatible? Meaning those it use the [UnReserved Characters](https://datatracker.ietf.org/doc/html/rfc3986#section-2.3) as shown at the link?

Why am I asking this? I want to implement a custom authentication flow with Auth0 SPA SDK and I need to generate  a high-entropy code verifier anc derive a challenge based on the previous code verifier and I wanted to use Nanoid to generate these.

Is it doable?

**Comment by ai** [comment, marked as answer]
Should be. Our alphabet `A-Za-z0-9_-`, seems like it inside `ALPHA / DIGIT / "-" / "." / "_" / "~"`
